elk/extraction/extraction.py

```diff
@@ -148,24 +135,69 @@ def extract_hiddens(
             )
             for layer_idx in layer_indices
         }
+        lm_preds = torch.empty(
+            num_variants,
+            2,  # contrast pair
+            device=device,
+            dtype=torch.float32,
+        )
         text_inputs = []
 
         # Iterate over variants
         for i, record in enumerate(example["prompts"]):
             variant_inputs = []
 
             # Iterate over answers
-            for j in range(2):
-                text = record[j]["text"]
+            for j, choice in enumerate(record):
+                text = choice["text"]
                 variant_inputs.append(text)
 
+                # TODO: Do something smarter than "rindex" here. Really we want to
+                # get the span of the answer directly from Jinja, but that doesn't
+                # seem possible. This approach may fail for complex templates.
+                answer_start = text.rindex(choice["answer"])
```

Comment:
I hope it's not necessary to include the answer in the question, because that would slow down inference (since if the answer only appears at the end you can cache the keys and values).

> I'm wondering if we should store the completion string (the second half of the jinja template) in the prompt dataset, and if it's empty and the user is attempting to use encoder-decoder models, we raise an exception.

Yes.

In general I'm like, yes, we should do the templates better, but I don't want that to hold up merging this PR